clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -113,6 +119,10 @@ public void changeSubscription(Collection<String> topicsToSubscribe) {
                 if (!subscription.contains(tp.topic()))
                     it.remove();
             }
+            
+            // Remove from pending subscribe those partitions that remain assigned to the consumer
```

Comment:
If we only add the unsubscribed partition in line 109 above, it looks we don't need line 123 to line 125 any more.